src/libpgmoneta/configuration.c

```diff
@@ -56,6 +56,7 @@
 
 static void extract_key_value(char* str, char** key, char** value);
 static int as_int(char* str, int* i);
+static int as_int_array(char* str, int* array, int max_len, int* len);
```

Comment:
Why do we need this ?

Reply:
Because input is in a form of the array, it's natural to parse it as an array and distribute the elements to the variables later. Plus I can do error handling there in one sitting.

Reply:
Rename the branch - do a new one without the array so that we can compare

Reply:
ok

Reply:
Just pushed a new PR, it only removes array in configuration.c, honestly I don't like it, at least for parsing array is much easier and simpler. I can try getting rid of other usage of array in this PR. But if you prefer the other one, I'll work on that then.